lib/errors/InvalidSourceError.js

```diff
@@ -1,23 +1,34 @@
-var NAME = "InvalidSourceError";
+var applyErrorPrototype = require("./applyErrorPrototype");
+
 /**
  * InvalidSourceError happens when a dataSource syncronously throws
  * an exception during a get/set/call operation.
  *
- * @param {Error} error - The error that was thrown.
+ * @param {Error} innerError - The error that was thrown.
+ *
  * @private
  */
-function InvalidSourceError(error) {
-    this.message = "An exception was thrown when making a request.";
-    this.stack = (new Error()).stack;
-    this.innerError = error;
+function InvalidSourceError(innerError) {
+    var instance = new Error("An exception was thrown when making a request.");
+
+    instance.name = "InvalidSourceError";
+    instance.innerError = innerError;
+
+    if (Object.setPrototypeOf) {
+        Object.setPrototypeOf(instance, Object.getPrototypeOf(this));
+    }
+
+    if (Error.captureStackTrace) {
+        Error.captureStackTrace(instance, InvalidSourceError);
+    }
+
+    return instance;
 }
 
-// instanceof will be an error, but stack will be correct because its defined
-// in the constructor.
-InvalidSourceError.prototype = new Error();
-InvalidSourceError.prototype.name = NAME;
+applyErrorPrototype(InvalidSourceError);
+
 InvalidSourceError.is = function(e) {
```

Comment:
this function can be deleted. `InvalidSourceError.is` is not used anywhere in source code.